pkg/kube/client.go

```diff
@@ -739,20 +748,20 @@ func (c *Client) pollCRDUntilEstablished(timeout time.Duration, info *resource.I
 			return false, fmt.Errorf("unable to get CRD: %v", err)
 		}
 
-		crd := &apiextv1beta1.CustomResourceDefinition{}
+		crd := &apiextv1.CustomResourceDefinition{}
 		err = scheme.Scheme.Convert(info.Object, crd, nil)
 		if err != nil {
 			return false, fmt.Errorf("unable to convert to CRD type: %v", err)
 		}
 
 		for _, cond := range crd.Status.Conditions {
 			switch cond.Type {
-			case apiextv1beta1.Established:
```

Comment:
Won't this break users who are still using the apiextensions/v1beta1 API? As far as I can remember, the business logic behind the two are slightly different, which is why we have `crdReady` and `crdBetaReady` in Helm 3. If so, this could be considered a breaking change for those users, as Helm will no longer wait for apiextensions/v1beta1 to become established.

Helm 3 handles both APIs. See here: https://github.com/helm/helm/blob/e31818f2efd3e1e0867d9c189507c25c8102264a/pkg/kube/wait.go#L111-L132

Reply:
Interesting, yeah I think you're right. We might be able to convert to `v1` from `v1beta1` via the internal type (there are conversions to/from `k8s.io/apiextensions-apiserver/pkg/apis/apiextensions` for both `v1` and `v1beta1`)

Reply:
I've refactored this to use unstructured directly instead. This seems to work nicely and supports both v1beta1 and v1.

Reply:
@bacongobbler Any thoughts on this approach?

Reply:
@bacongobbler Sorry to pester you but would you mind re-reviewing please?